extensions/wikia/Email/Controller/WallMessageController.class.php

```diff
@@ -0,0 +1,111 @@
+
+use Email\EmailController;
+
+class WallMessageController extends EmailController {
+
+	/**
+	 * @template wallMessage
+	 */
+	public function body() {
+		$this->response->setData( [
+			'salutation' => $this->getSalutation(),
+			'summary' => $this->getSummary(),
+			'editorProfilePage' => $this->getCurrentProfilePage(),
+			'editorUserName' => $this->request->getVal( 'fromName' ),
+			'editorAvatarURL' => $this->getCurrentAvatarURL(),
+			'wallMessageTitle' => $this->request->getVal( 'messageTitle' ),
+			'wallMessageBody' => $this->request->getVal( 'messageBody' ),
+			'buttonText' => $this->getFullConversationText(),
+			'buttonLink' => $this->getFullConversationUrl(),
+			'contentFooterMessages' => [
+				$this->getRecentMessagesText()
+			],
+		] );
+	}
+
+	/**
+	 * Whether this email is for a user following a thread on someone else's wall
+	 *
+	 * @return boolean
+	 */
+	protected function isForFollowedThread() {
+		return $this->request->getVal( 'targetUser' ) !== $this->request->getVal( 'wallUserName' );
+	}
+
+	/**
+	 * Get the email subject line
+	 *
+	 * @return string
+	 */
+	public function getSubject() {
+		if ( $this->isForFollowedThread() ) {
+			return wfMessage( 'emailext-wallmessage-following-subject',
+				$this->request->getVal( 'fromName' ),
+				$this->request->getVal( 'wallUserName' )
+			)->inLanguage( $this->targetLang )->text();
+		} else {
+			return wfMessage( 'emailext-wallmessage-owned-subject',
+				$this->request->getVal( 'fromName' )
+			)->inLanguage( $this->targetLang )->text();
+		}
+	}
+
+	/**
+	 * Get the summary text immediately following the salutation in the email
+	 *
+	 * @return string
+	 */
+	protected function getSummary() {
+		if ( $this->isForFollowedThread() ) {
+			return wfMessage( 'emailext-wallmessage-following-summary',
+				$this->request->getVal( 'fromName' ),
+				$this->request->getVal( 'wallUserName' )
+			)->inLanguage( $this->targetLang )->parse();
+		} else {
+			return wfMessage( 'emailext-wallmessage-owned-summary',
+				$this->request->getVal( 'fromName' )
+			)->inLanguage( $this->targetLang )->parse();
+		}
+	}
+
+	/**
+	 * Get the localized text describing the full message thread link
+	 *
+	 * @return string
+	 */
+	protected function getFullConversationText() {
+		return wfMessage( 'emailext-wallmessage-full-conversation' )->inLanguage( $this->targetLang )->text();
+	}
+
+	/**
+	 * Get the URL to the full message thread
+	 *
+	 * @return string
+	 */
+	protected function getFullConversationUrl() {
+		return $this->request->getVal( 'url' );
+	}
+
+	/**
+	 * Get the localized text describing the recent messages link
+	 *
+	 * @return string
+	 */
+	protected function getRecentMessagesText() {
+		$wallTitleText = $this->isForFollowedThread() ? $this->request->getVal( 'wallUserName' ) :
+			$this->request->getVal( 'targetuser' );
+		return wfMessage( 'emailext-wallmessage-recent-messages',
+			\Wall::newFromTitle( \Title::newFromText( $wallTitleText ) )->getUrl()
```

Comment:
One note here; you should be careful with the one argument form of ```Title::newFromText```.  The second argument is the namespace which defaults to ```NS_MAIN``` (e.g. /wiki/SomeTitle), so if you're trying to load a title that isn't in the ```NS_MAIN``` namespace (e.g., a file, /wiki/File:SomeTitle), you'll end up either:
1) Loading the wrong Title object if there is a Title with the same name in the ```NS_MAIN``` namespace
2) Loading an empty Title object if there isn't a Title with the same name.
In the case above its actually #2 since you're trying to load a Title in the ```NS_USER_WALL``` namespace (e.g., /wiki/Message_Wall:Garthwebb), however ```Wall::newFromTitle``` *happens* to make calls on the empty Title object that don't make DB requests and  essentially returns it the ```$wallTitleText``` you passed in to construct the proper URL.